src/App/src/Handler/AbstractHandler.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Api\App\Handler;
+
+use Api\App\Collection\CollectionInterface;
+use Api\App\Entity\EntityInterface;
+use Api\App\Exception\BadRequestException;
+use Api\App\Exception\ConflictException;
+use Api\App\Exception\ExpiredException;
+use Api\App\Exception\ForbiddenException;
+use Api\App\Exception\MethodNotAllowedException;
+use Api\App\Exception\NotFoundException;
+use Api\App\Exception\RuntimeException;
+use Api\App\Exception\UnauthorizedException;
+use Api\App\Message;
+use Dot\Mail\Exception\MailException;
+use Exception;
+use Fig\Http\Message\StatusCodeInterface;
+use Laminas\Diactoros\Response\EmptyResponse;
+use Laminas\Diactoros\Response\JsonResponse;
+use Mezzio\Hal\HalResponseFactory;
+use Mezzio\Hal\ResourceGenerator;
+use Mezzio\Hal\ResourceGenerator\Exception\OutOfBoundsException;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+use function assert;
+use function is_array;
+use function method_exists;
+use function strtolower;
+
+abstract class AbstractHandler implements RequestHandlerInterface
```

Comment:
Good point! 👍 But...
I made `AbstractHandler` an abstract class because it should not be instantiatable on its own, only extended by other handler classes. It is not a standalone class, it only provides common functionalities to be consumed by other handlers.

I had to look this up just to make sure I remember correcty, but having at least one abstract method is not mandatory for a class to be abstract. It's quite the opposite: if a class has at least one abstract method, then it must be declared abstract.

See the first paragraph of the [PHP documentation on abstract classes](https://www.php.net/manual/en/language.oop5.abstract.php):

>  PHP has abstract classes and methods. Classes defined as abstract cannot be instantiated, and any class that contains at least one abstract method must also be abstract. Methods defined as abstract simply declare the method's signature; they cannot define the implementation.